blivet/formats/fs.py

```diff
@@ -317,32 +315,25 @@ def update_size_info(self):
         self._min_instance_size = Size(0)
         self._resizable = self.__class__._resizable
 
-        # We can't allow resize if the filesystem has errors.
+        # try to gather current size info
+        self._size = Size(0)
         try:
-            self.do_check()
-        except FSError:
+            if self._info.available:
+                self._current_info = self._info.do_task()
+        except FSError as e:
+            log.info("Failed to obtain info for device %s: %s", self.device, e)
+        try:
+            self._size = self._size_info.do_task()
+            self._min_instance_size = self._size
+        except (FSError, NotImplementedError) as e:
+            log.warning("Failed to obtain current size for device %s: %s", self.device, e)
+
+        # We absolutely need a current size to enable resize. To shrink the
```

Comment:
I'm not completely sure this is true, but it's probably a good indicator of having the tools and the state required to resize.